api/compose/api.go

```diff
@@ -254,6 +255,9 @@ func (opts *RunOptions) EnvironmentMap() types.MappingWithEquals {
 		switch {
 		case len(parts) == 1:
 			environment[key] = nil
+			if v, ok := os.LookupEnv(key); ok {
```

Comment:
don't use `os.LookupEnv` but `project.Environment` which combine all possible sources for variables (.env, os.Env, --env-file...) and can be extended/better controlled in the future

Also, consider using `MappingWithEquals.Resolve`

Reply:
see https://github.com/docker/compose-cli/pull/1610/files#diff-477fda25fb05d92fe2c8e3cb31fdc228a8f5597b9e26b22f7feeff8246404db7R45-R51 as an example you can reuse in `local/compose/run.go:RunOneOffContainer`

Reply:
I just noticed this whole method could be simplified as it just reproduce existing `NewMappingWithEquals`